application/src/main/java/org/thingsboard/server/service/install/update/DefaultDataUpdateService.java

```diff
@@ -681,4 +691,69 @@ private boolean getEnv(String name, boolean defaultValue) {
         }
     }
 
+    private final PaginatedUpdater<String, Tenant> tenantsDashboardsFiltersUpdater =
+            new PaginatedUpdater<>() {
+
+                @Override
+                protected String getName() {
+                    return "Tenants dashboards filters updater";
+                }
+
+                @Override
+                protected boolean forceReportTotal() {
+                    return true;
+                }
+
+                @Override
+                protected PageData<Tenant> findEntities(String region, PageLink pageLink) {
+                    return tenantService.findTenants(pageLink);
+                }
+
+                @Override
+                protected void updateEntity(Tenant tenant) {
+                    updateTenantDashboardsFilters(tenant.getId());
+                }
+            };
+
+    private void updateTenantDashboardsFilters(TenantId tenantId) {
+        PageLink pageLink = new PageLink(100);
+        PageData<DashboardInfo> pageData = dashboardService.findDashboardsByTenantId(tenantId, pageLink);
```

Comment:
This approach will be extremely slow in production. You should search the dashboards that have the corresponding filter (based on the full text search, etc) and update only the ones that are needed. We have 50K tenants and maybe 1K dashboards that use this feature on demo.thingsboard.io.